lib/internal/worker.js

```diff
@@ -110,7 +111,13 @@ class Worker extends EventEmitter {
         );
       }
       url = null;
+      doEval = 'classic';
+    } else if (filename?.protocol === 'data:') {
+      url = null;
+      doEval = 'module';
+      filename = `import ${JSONStringify(filename)}`;
```

Comment:
In order to ensure we are generating a module with a string in this location, we should coerce to string before stringify:

```suggestion
      filename = `import ${JSONStringify(`${filename}`)}`;
```

This really only affects if someone defines a custom `.toJSON` on the prototype chain somewhere since we already do a URL instance check.